clients/drcachesim/simulator/cache_simulator.h

```diff
@@ -60,12 +60,26 @@ class cache_simulator_t : public simulator_t {
     bool
     print_results() override;
 
+    int_least64_t
+    get_l1i_metric(metric_name_t metric, unsigned int core) const;
+    int_least64_t
+    get_l1d_metric(metric_name_t metric, unsigned int core) const;
+    int_least64_t
+    get_ll_metric(metric_name_t metric) const;
+
     // Exposed to make it easy to test
     bool
     check_warmed_up();
     uint64_t
     remaining_sim_refs() const;
 
+    unsigned int
+    get_core_num() const;
```

Comment:
Since cache_simulator_knobs_t is already public -- should the knobs be returned?  IIRC a config file setup does set the key knobs fields?

Reply:
Yes, knobs fields are set by config file setup. I have added suitable method.